src/streamlink_cli/main.py

```diff
@@ -123,7 +127,19 @@ def create_http_server(host=None, port=0):
 
     return http
 
+def create_title(plugin=None):
+    if args.title and plugin:
+        title = args.title.format(title=plugin.get_title(),
```

Comment:
If possible this should only call the methods that are required to build the title, otherwise there could be slowdowns from extra API calls, etc.

Reply:
Might be nice, if they all returned `None` (previous comment), to default back to `args.url` for the title.